lib.py

```diff
@@ -1,167 +1,284 @@
+            environment={
+                "FERRIS_ELF_INPUT_FILE_NAME": in_file_name,
+            },
+            remove=True,
+            stdout=True,
+            stderr=True,
+            mem_limit="8g",
+            # network_mode="none", # Downloading crates.
+            volumes={tmp_dir: {'bind': '/app', 'mode': 'rw'}}
+        ))
+        out: str = out.decode("utf-8")
+        logger.debug("Run container output (type: %s):\n%s", type(out), out)
+        results = []
 
-    db = Database().get()
+        for l in out.splitlines():
+            if len(l) == 0 or l[0] != '{':
+                continue
+            l_data = json.loads(l)
+            results.append(l_data)
 
-    verified = False
-    results = []
-    for i, file in enumerate(onlyfiles):
-        rows = db.cursor().execute(
-            "SELECT answer2 FROM solutions WHERE key = ? AND day = ? AND part = ?",
-            (file, day, part),
-        )
+        logger.debug("Results from container run for user %s, file %s: %s", author_id, in_file, results)
+        return results
+    except docker.errors.ContainerError:
+        logger.exception("Error in docker while running code")
+        return None
 
-        with open(os.path.join(day_path, file), "r") as f:
-            input = f.read()
-
-        status = await msg.reply(f"Benchmarking input {i+1}")
-        out = await run_image(msg, input)
-        if not out:
-            return
-        await status.delete()
-
-        result = {}
-        # TODO: get the answer and the bench results from the docker and verify them here
-
-        cur.execute(
-            "INSERT INTO runs VALUES (?, ?, ?, ?, ?, ?, ?)",
-            (
-                str(msg.author.id),
-                code,
-                day,
-                part,
-                result["median"],
-                result["answer"],
-                result["answer"],
-            ),
-        )
-        results.append(result)
+def process_run_result(in_file, answers_map, result_lst):
+    """Given JSON blobs extracted from a container's stdout, get the core stats out."""
+    # `result` should be a dataclass, not a dict. Another time.
+    result = {}
+    for blob in result_lst:
+        reason = blob.get("reason", None)
+        if reason == None:
+            continue
+        elif reason == "ferris-answer":
+            answer = blob["answer"]
+            result["answer"] = answer
+            if answers_map.get(in_file, None) == answer:
+                result["verified"] = True
+            else:
+                result["verified"] = False
+        elif reason == "benchmark-complete":
+            result["typical"] = blob["typical"]["estimate"]
+            result["average"] = blob["mean"]["estimate"]
+            result["median"] = blob["median"]["estimate"]
+            result["high_bound"] = blob["typical"]["upper_bound"]
+            result["low_bound"] = blob["typical"]["lower_bound"]
```

Comment:
This should probably be a dataclass, it also improves typeck on this portion of code
```py
@dataclass(slots=True)
class Result:
    typical: float
    average: float
    median: float
    high_bound: float
    low_bound: float
```